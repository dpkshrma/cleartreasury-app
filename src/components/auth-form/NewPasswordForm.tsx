import React, { FunctionComponent } from "react";
import { Button, Input } from "@clear-treasury/design-system";
import { FieldErrors, useFormContext } from "react-hook-form";
import { FormType } from "./types";

export interface NewPasswordFormProps {
  errors?: FieldErrors;
  loading?: boolean;
  formType?: FormType;
}

export const NewPasswordForm: FunctionComponent<NewPasswordFormProps> = ({
  loading,
  errors,
  formType,
}) => {
  if (formType !== FormType.newPasswordForm) return null;

  const formContext = useFormContext();
  let emailInputProps = {};
  let passwordInputProps = {};
  if (formContext) {
    emailInputProps = formContext.register("email");
    passwordInputProps = formContext.register("password");
  }

  return (
    <>
      <Input
        type="email"
        name="email"
        label="Email address"
        placeholder="Enter your email"
        errors={errors}
        {...emailInputProps}
      />
      <Input
        type="password"
        name="newPassword"
        label="Change your password"
        placeholder="Enter your new password"
        errors={errors}
        {...passwordInputProps}
      />

      <Button size={Button.Size.LARGE} loading={loading}>
        Set new password
      </Button>
    </>
  );
};
