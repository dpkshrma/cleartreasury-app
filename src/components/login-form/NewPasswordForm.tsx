import React, { Ref } from "react";
import { Button, Input } from "@clear-treasury/design-system";

export interface NewPasswordFormProps {
  passwordRef: Ref<HTMLInputElement | null>;
  errors: any;
  loading: boolean;
}

export const NewPasswordForm: React.FunctionComponent<NewPasswordFormProps> = ({
  passwordRef,
  loading,
  errors,
}) => (
  <>
    <Input
      type="password"
      name="newPassword"
      label="Change your password"
      placeholder="Enter your new password"
      ref={passwordRef}
      errors={errors}
    />

    <Button size={Button.Size.LARGE} loading={loading}>
      Set new password
    </Button>
  </>
);
