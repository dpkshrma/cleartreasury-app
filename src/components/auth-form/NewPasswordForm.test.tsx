import * as React from "react";
import { render } from "../../../test/testUtils";
import { Default } from "./NewPasswordForm.stories";
import { fireEvent, screen } from "@testing-library/react";
import { act } from "react-dom/test-utils";

jest.mock("@aws-amplify/auth");

const inputEmail = (value: string) =>
  act(async () => {
    fireEvent.input(screen.getByRole("textbox", { name: /email address/i }), {
      target: { value },
    });
  });
const inputPassword = (value: string) =>
  act(async () => {
    fireEvent.input(screen.getByLabelText(/change your password/i), {
      target: { value },
    });
  });
const submitForm = () =>
  act(async () => {
    fireEvent.click(screen.getByRole("button", { name: /set new password/i }));
  });

describe("SignInForm", () => {
  it("should match snapshot", () => {
    const { asFragment } = render(<Default {...Default.args} />);
    expect(asFragment()).toMatchSnapshot();
  });

  it("should call onSubmit prop on form submit", async () => {
    const onSubmit = jest.fn();
    render(<Default {...Default.args} onSubmit={onSubmit} />);
    await inputEmail("test@example.com");
    await inputPassword("password");
    await submitForm();
    expect(onSubmit).toBeCalled();
  });

  it("should not call onSubmit prop on empty form submit", async () => {
    const onSubmit = jest.fn();
    render(<Default {...Default.args} onSubmit={onSubmit} />);
    await submitForm();
    expect(screen.queryByText("email is a required field")).toBeTruthy();
    expect(screen.queryByText("newPassword is a required field")).toBeTruthy();
    expect(onSubmit).not.toBeCalled();
  });

  it("should check for valid email address", async () => {
    const onSubmit = jest.fn();
    render(<Default {...Default.args} onSubmit={onSubmit} />);
    await inputEmail("test");
    await inputPassword("password");
    await submitForm();
    expect(screen.queryByText("email must be a valid email")).toBeTruthy();
    expect(onSubmit).not.toBeCalled();
  });
});
