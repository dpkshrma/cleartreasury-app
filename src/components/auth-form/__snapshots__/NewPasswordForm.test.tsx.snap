// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`SignInForm should match snapshot 1`] = `
<DocumentFragment>
  <form
    class="flex justify-center flex-col space-y-6"
  >
    <h1
      class="block w-full text-center text-gray-800 text-2xl"
    >
      Set your password
    </h1>
    <div
      class="w-full relative "
    >
      <label
        class="text-base text-gray-700 inline-block mb-2"
        for="email"
        id="email-label"
      >
        Email address
      </label>
      <input
        class="form-input border border-gray-300 bg-white mb-2 p-3 w-full focus:ring-teal-100 focus:border-teal-300 disabled:cursor-not-allowed placeholder-gray-400 rounded"
        id="email"
        name="email"
        placeholder="Enter your email"
        type="email"
        value=""
      />
    </div>
    <div
      class="w-full relative "
    >
      <label
        class="text-base text-gray-700 inline-block mb-2"
        for="newPassword"
        id="newPassword-label"
      >
        Change your password
      </label>
      <input
        class="form-input border border-gray-300 bg-white mb-2 p-3 w-full focus:ring-teal-100 focus:border-teal-300 disabled:cursor-not-allowed placeholder-gray-400 rounded"
        id="newPassword"
        name="newPassword"
        placeholder="Enter your new password"
        type="password"
        value=""
      />
    </div>
    <button
      class="inline-flex items-center justify-center gap-2
    border border-transparent
    ring-theme-button-primary
    text-center leading-snug
    rounded-theme-radius
    focus:outline-none focus:ring-1 focus:ring-offset-1 focus:ring-offset-transparent
    transition-all duration-300
    hover:shadow-md hover:cursor-pointer bg-theme-button-primary text-theme-color-on-primary hover:bg-theme-button-primary-hover py-3 px-8"
    >
      Set new password
    </button>
  </form>
</DocumentFragment>
`;
